### DISCUSSION TITLE
Build the milvus from source

### DISCUSSION THREAD

**Discussion by chen982** [milvus-io/milvus#26072] [Q&A and General discussion] [answered]
I followed the guild for make milvus and got a bin/milvus executable file, but when I run this, it shows something error。Does someone know how to fix it ?
(base) ./milvus 
./milvus: error while loading shared libraries: libabsl_time.so.2206.0.0: cannot open shared object file: No such file or directory

**Comment by yhmo** [comment, marked as answer]
1. cmd to the project root path
2. `source ./scripts/setenv.sh`
3. `cd deployments/docker/dev`
4. `docker-compose up -d`
5. `cd -`
6.  `./scripts/start_standalone.sh`

**Reply by chen982** [reply]
Thanks for your kindly reply, it's helpful

**Reply by chen982** [reply]
It seems can be run in normal user, but finally I will get panic: IO error: while open a file for lock: /var/lib/milvus/rdb_data_meta_kv/LOCK: Permission denied. 
when use sudo to execute it , the user env seems not to be reserved, and it again shows ./milvus: error while loading shared libraries: libabsl_time.so.2206.0.0: cannot open shared object file: No such file or directory

**Reply by yhmo** [reply]
Sorry, miss a step.

`./scripts/stop.sh`
Replace "/var/lib/milvus" with "/tmp/milvus" in the `configs/milvus.yaml`  
`./scripts/start_standalone.sh`

**Reply by yhmo** [reply]
Under the `deployments/docker/dev` there is a "volumes" folder. This folder stores all the data of etcd/minio.
Sometimes the source code changed, the local volumes might contain legacy data and the server failed to start. Then you can clean all the legacy data so that you can start a clean server:

`./scripts/stop.sh`
`cd deployments/docker/dev`
`sudo rm -rf ./volumes`

**Reply by chen982** [reply]
Done! Thanks so much for your kindly reply!
